File: backend/internal/models/production.go
package models

import (
	"time"
	"gorm.io/gorm"
)

// ProductionOrder 生产工单
type ProductionOrder struct {
	ID           uint           `json:"id" gorm:"primarykey"`
	OrderNo      string         `json:"order_no" gorm:"uniqueIndex;size:50;not null"`
	ProductID    uint           `json:"product_id"`
	Product      Product        `json:"product" gorm:"foreignKey:ProductID"`
	Quantity     int            `json:"quantity" gorm:"not null"`
	Produced     int            `json:"produced" gorm:"default:0"`
	Status       string         `json:"status" gorm:"size:20;default:'pending'"` // pending, processing, completed, cancelled
	Priority     int            `json:"priority" gorm:"default:1"`
	StartDate    *time.Time     `json:"start_date"`
	EndDate      *time.Time     `json:"end_date"`
	CreatedBy    uint           `json:"created_by"`
	Creator      User           `json:"creator" gorm:"foreignKey:CreatedBy"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
	DeletedAt    gorm.DeletedAt `json:"-" gorm:"index"`
}

// Product 产品信息
type Product struct {
	ID          uint           `json:"id" gorm:"primarykey"`
	Code        string         `json:"code" gorm:"uniqueIndex;size:50;not null"`
	Name        string         `json:"name" gorm:"size:100;not null"`
	Description string         `json:"description" gorm:"type:text"`
	Unit        string         `json:"unit" gorm:"size:20"`
	Price       float64        `json:"price" gorm:"type:decimal(10,2)"`
	Status      int            `json:"status" gorm:"default:1"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

// TableName 指定表名
func (ProductionOrder) TableName() string {
	return "production_orders"
}

func (Product) TableName() string {
	return "products"
}

File: frontend/src/api/production.ts
import apiClient from './client';
import type { ApiResponse, PageResponse } from '../types';

export interface ProductionOrderRequest {
  product_id: number;
  quantity: number;
  priority: number;
  start_date: string;
  end_date: string;
  description?: string;
}

export interface ProductionOrderResponse {
  id: number;
  order_no: string;
  product_id: number;
  product?: {
    id: number;
    code: string;
    name: string;
    unit: string;
  };
  quantity: number;
  produced: number;
  priority: number;
  status: string;
  start_date: string;
  end_date: string;
  created_by: number;
  creator?: {
    id: number;
    username: string;
    real_name: string;
  };
  created_at: string;
  updated_at: string;
}

export interface ProductionOrderQueryParams {
  page?: number;
  page_size?: number;
  status?: string;
  product_id?: number;
}

/**
 * 获取生产工单列表
 */
export const getProductionOrders = (params?: ProductionOrderQueryParams) =>
  apiClient.get<ApiResponse<PageResponse<ProductionOrderResponse>>>('/production/orders', { params });

/**
 * 创建生产工单
 */
export const createProductionOrder = (data: ProductionOrderRequest) =>
  apiClient.post<ApiResponse<ProductionOrderResponse>>('/production/orders', data);

/**
 * 获取生产工单详情
 */
export const getProductionOrderById = (id: number) =>
  apiClient.get<ApiResponse<ProductionOrderResponse>>(`/production/orders/${id}`);

/**
 * 更新生产工单
 */
export const updateProductionOrder = (id: number, data: ProductionOrderRequest) =>
  apiClient.put<ApiResponse<ProductionOrderResponse>>(`/production/orders/${id}`, data);

/**
 * 删除生产工单
 */
export const deleteProductionOrder = (id: number) =>
  apiClient.delete<ApiResponse<any>>(`/production/orders/${id}`);

/**
 * 获取生产工单状态列表
 */
export const getProductionOrderStatuses = () =>
  apiClient.get<ApiResponse<string[]>>('/production/orders/statuses');